### PR TITLE
KL-580: add support for signTypedData methods

### PR DESCRIPTION
**Pull request by Boltman92**

### CODE REVIEW

File: libs/ui/src/modals/screens/d-app-sign-confirmation/d-app-sign-confirmation.tsx [hunk 1]
```diff
@@ -29,12 +29,32 @@ export const DAppSignConfirmation: FC = () => {
 
   const onDecline = () => sendErrorToDAppAndClosePopup(params.dAppInfo.origin, params.messageId);
 
-  const messageToSign = params.method === DAppMethodEnum.ETH_PERSONAL_SIGN ? params.signInfo[0] : params.signInfo[1];
+  const messageToSign = (method: string) => {
```
Comment (unixvb) [inline libs/ui/src/modals/screens/d-app-sign-confirmation/d-app-sign-confirmation.tsx]: move this into d-app-sign-confirmation/utils

File: libs/ui/src/modals/screens/d-app-sign-confirmation/d-app-sign-confirmation.tsx [hunk 2]
```diff
@@ -52,7 +72,7 @@ export const DAppSignConfirmation: FC = () => {
       <DAppHeader favicon={params.dAppInfo.favicon} origin={params.dAppInfo.origin} />
       <View style={styles.messageBlock}>
         <Text style={styles.mainText}>Message to sign</Text>
-        <Text style={styles.text}>{prepareSignData(messageToSign)}</Text>
+        <Text style={styles.text}>{prepareSignData(messageToSign(params.method))}</Text>
```
Comment (unixvb) [inline libs/ui/src/modals/screens/d-app-sign-confirmation/d-app-sign-confirmation.tsx]: `prepareSignData(messageToSign(params.method))` duplication.  
create variable and reuse it

File: libs/ui/src/shelter/utils/sign-message-by-method.util.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
+import { signTypedData, SignTypedDataVersion } from '@metamask/eth-sig-util';
+import { ethers } from 'ethers';
+import { of } from 'rxjs';
+
+import { DAppMethodEnum } from '../../enums/dApp-method.enum';
+
+export function signMessageByMethod(privateKey: string, messageToSign: string, method: string) {
```
Comment (unixvb) [inline libs/ui/src/shelter/utils/sign-message-by-method.util.ts]: arrow function pls

File: libs/ui/src/shelter/utils/sign-message-by-method.util.ts [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    case DAppMethodEnum.ETH_SIGN_TYPED_DATA:
+    case DAppMethodEnum.ETH_SIGN_TYPED_DATA_V3:
+    case DAppMethodEnum.ETH_SIGN_TYPED_DATA_V4: {
+      const version = method.split('_')[2] ? method.split('_')[2].toUpperCase() : SignTypedDataVersion.V1;
+      const data = JSON.parse(messageToSign);
+      const privateKeyBuffer = Buffer.from(privateKey.slice(2), 'hex');
+      const signedMessage = signTypedData({
+        privateKey: privateKeyBuffer,
+        data,
+        version: version as SignTypedDataVersion
```
Comment (unixvb) [inline libs/ui/src/shelter/utils/sign-message-by-method.util.ts]: could you remove assertion?

**Review by thesvyatoslavtt — DISMISSED** [review]

**Review by unixvb — DISMISSED** [review]

**Review by vladvn1905 — DISMISSED** [review]

**Review by unixvb — APPROVED** [review]

**Pull request merged**